lib/elixir/lib/process.ex

```diff
@@ -421,6 +421,12 @@ defmodule Process do
   @spec register(pid | port, atom) :: true
   def register(pid, name) when not name in [nil, false, true] and is_atom(name) do
     :erlang.register(name, pid)
+  catch
```

Comment:
@fishcakez Good catch. Let's merge this without the guard changes because it doesn't  feel like we should be registering ports through here while there is a `Port` module.